### ISSUE TITLE
jsPsych preloading video blocked due to CORS

### ISSUE THREAD

**Issue by becky-gilbert** [lookit/lookit-api#1486]
## TL;DR

Video preloading produces a CORS error.

## Summary

In CHS jsPsych studies, it looks like attempts to preload video files with the jsPsych `preload` plugin are being blocked by the site's CORS policy. For instance, attempting to preload like this:

```js
var preload = {
    type: jsPsychPreload,
    video: ["https://github.com/nicolehope5/Sesame-Street-Functions-stimuli/raw/refs/heads/main/mp4/Movie1_Intro.mp4"]
}
```

Produces the following network error: "CORS Allow Origin Not Matching Origin". 

Whereas there's no CORS error when the stimuli URLs are used by plugins directly, e.g. as parameters in plugins or written into the plugin HTML.

Interestingly, there's no CORS error when preloading this image file from the same domain:

```js
var preload = {
    type: jsPsychPreload,
    images: ["https://github.com/nicolehope5/Sesame-Street-Functions-stimuli/raw/refs/heads/main/img/purplearrow.png"]
}
```

All of the preloading is done via the jsPsych plugin API functions. It looks like the jsPsych MediaAPI uses a different method for [preloading video](https://github.com/jspsych/jsPsych/blob/b0c6736b166fbb962dba8953d312c5adc4932b2e/packages/jspsych/src/modules/plugin-api/MediaAPI.ts#L143-L192) vs [preloading images](https://github.com/jspsych/jsPsych/blob/b0c6736b166fbb962dba8953d312c5adc4932b2e/packages/jspsych/src/modules/plugin-api/MediaAPI.ts#L105-L141), so that might be relevant.

## Screenshots

Video preloading error (initiated by jsPsych core, i.e. plugin API preload function):

![Screenshot 2024-11-25 at 1 19 12 PM](https://github.com/user-attachments/assets/b445c3d3-0b41-44ec-9a64-807ca78d92e8)

![Screenshot 2024-11-25 at 1 15 38 PM](https://github.com/user-attachments/assets/b01d4fdc-696f-4d5f-a6ee-041fce1811f3)

Images preload successfully (initiated by jsPsych core, i.e. plugin API preload function):

![Screenshot 2024-11-25 at 1 18 00 PM](https://github.com/user-attachments/assets/5c483a2f-790c-48f5-80dc-ce5b3d6442c4)

Video loads successfully during the trial (initiated by plugin HTML):

![Screenshot 2024-11-25 at 1 13 10 PM](https://github.com/user-attachments/assets/333bcbd4-5dc0-45dd-a30c-bb59e396d503)

**Comment by okaycj** [comment]
@becky-gilbert It might be possible that jsPsych isn't following redirects (302).  The above URL will redirect to a different domain "raw.githubusercontent.com".  For now we could just use the URL `https://raw.githubusercontent.com/nicolehope5/Sesame-Street-Functions-stimuli/refs/heads/main/mp4/Movie1_Intro.mp4` instead.

**Comment by becky-gilbert** [comment]
@okaycj is exactly right - the problem was that jsPsych was not following redirects (at least for videos). This problem was fixed by changing the links provided to the `preload` plugin so that they did not redirect. I'll also see if this can be addressed by jsPsych, and in the meantime I'll make sure to make a note of this in our documentation.
